core/MyGatewayTransportMQTTClient.cpp

```diff
@@ -273,6 +273,11 @@ bool gatewayTransportAvailable(void)
 	if (_MQTT_connecting) {
 		return false;
 	}
+#if defined(MY_GATEWAY_ESP8266) || defined(MY_GATEWAY_ESP32)
+	if (WiFi.status() != WL_CONNECTED) {
+		return false;
```

Comment:
ESP32 won't reconnect unless you call gatewayTransportInit();

Reply:
Ok, reworked based on your comments